automatic/zotero/update.ps1

```diff
@@ -11,32 +10,25 @@ function global:au_SearchReplace {
 
   @{
     ".\legal\VERIFICATION.txt" = @{
-      "(?i)(\s*1\..+)\<.*\>" = "`${1}<$($Latest.URL32)>"
-      "(?i)(^\s*checksum\s*type\:).*" = "`${1} $($Latest.ChecksumType32)"
-      "(?i)(^\s*checksum(32)?\:).*" = "`${1} $($Latest.Checksum32)"
-    }
-    ".\tools\chocolateyInstall.ps1" = @{
-      "(?i)^(\s*softwareName\s*=\s*)'.*'" = "`${1}'$softwareName'"
-      "(?i)(^\s*file\s*=\s*`"[$]toolsPath\\).*" = "`${1}$($Latest.FileName32)`""
-    }
-    ".\tools\chocolateyUninstall.ps1" = @{
-      "(?i)^(\s*softwareName\s*=\s*)'.*'" = "`${1}'$softwareName'"
-    }
-    ".\$($Latest.PackageName).nuspec" = @{
-      "(\<releaseNotes\>).*(\<\/releaseNotes\>)" = "`${1}https://www.zotero.org/support/$($version.Major).$($version.Minor)_changelog`${2}"
+      "(?i)(^\s*32-bit download\:).*" = "`${1} $($Latest.URL32)"
+      "(?i)(^\s*64-bit download\:).*" = "`${1} $($Latest.URL64)"
+      "(?i)(^\s*32-bit checksum\:).*" = "`${1} $($Latest.Checksum32)"
+      "(?i)(^\s*64-bit checksum\:).*" = "`${1} $($Latest.Checksum64)"
     }
   }
 }
 
 function global:au_GetLatest {
-    $url = Get-RedirectedUrl -url $releases
+   $url64 = Get-RedirectedUrl -url $releases
+   $url32 = $url64 -replace 'x64','win32'
 
-    $version  = $url -split '/' | Select-Object -Last 1 -Skip 1
+   $version  = $url64.split('/') | Where-Object {$_ -match '^[0-9.]+$'}
```

Comment:
This regex may cause false positives, as well as can cause multiple matches to be found if the URLs are ever changed.

It is better that the update fails, than us having false/duplicated matches.

Reply:
The previous method of grabbing the version could just as easily cause a failure if the URL changes.  I believe the updater would fail if `$version` is an array too.  Either way, we are dependent on the publisher to be consistent, and I don't see how one version is better than the other.

Reply:
We want to prevent false positives from happening as much as possible, it is fine when something fails, but a false positive may cause a package to be pushed with the wrong information.

In the regards of an arary of versions failing (or numbers in this case), that may be the case, but it isn't a guarantee. AFAIK, this hasn't been checked to see if it is the case or not.

Reply:
How about:

    $version  = $url64 -split '/' | Where-Object {$_ -match '^\d+\.\d[0-9.]+$'} | Select-Object -Last 1

That ensures the value is "version-like" and there is just one returned.